Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift

```diff
@@ -5,169 +5,92 @@
 import StreamChat
 import UIKit
 
-open class MessageComposerSuggestionsViewController<ExtraData: ExtraDataTypes>: UIViewController,
-    UITableViewDelegate,
-    UITableViewDataSource {
+open class MessageComposerSuggestionsViewController<ExtraData: ExtraDataTypes>: ViewController,
+    UIConfigProvider,
+    UICollectionViewDelegate,
+    UICollectionViewDataSource {
+    var heightConstraint: NSLayoutConstraint?
+
     // MARK: - Property
 
-    public let uiConfig: UIConfig<ExtraData>
-    
-    public var suggestions: [String] = ["Brian", "David", "Pavel"] {
-        didSet {
-            updateContent()
-        }
-    }
-    
-    public var position: CGPoint = CGPoint(x: 0, y: 300) {
-        didSet {
-            updateContent()
-        }
-    }
-    
-    public var maxHeight: CGFloat = 300.0 {
-        didSet {
-            updateContent()
-        }
-    }
-    
-    public var calculatedHeight: CGFloat {
-        min(CGFloat(40 * suggestions.count), maxHeight)
-    }
-    
+    private var collectionViewHeightObserver: NSKeyValueObservation?
+
     // MARK: - Subviews
-    
-    public private(set) lazy var tableView = UITableView().withoutAutoresizingMaskConstraints
-    
-    // MARK: - Init
-    
-    public required init(
-        uiConfig: UIConfig<ExtraData> = .default
-    ) {
-        self.uiConfig = uiConfig
-
-        super.init(nibName: nil, bundle: nil)
-        
-        commonInit()
-    }
-    
-    public required init?(coder: NSCoder) {
-        uiConfig = .default
-        
-        super.init(coder: coder)
-        
-        commonInit()
-    }
-    
-    public func commonInit() {
-        tableView.dataSource = self
-        tableView.delegate = self
-        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
-        modalPresentationStyle = .custom
-        modalTransitionStyle = .crossDissolve
-    }
-    
+
+    public private(set) lazy var collectionView = uiConfig
+        .messageComposer
+        .suggestionsCollectionView
+        .init(layout: uiConfig.messageComposer.suggestionsCollectionViewLayout.init())
+        .withoutAutoresizingMaskConstraints
+
```

Comment:
Please use `open` by default unless something is mandatory to not be overridden (like `defaultAppearance` for example)